l10n_it_intrastat_statement/models/intrastat_statement.py

```diff
@@ -768,17 +768,28 @@ def compute_statement(self):
         period_date_start, period_date_stop = self.get_dates_start_stop()
 
         # Search intrastat lines
-        domain = [
-            ("date", ">=", period_date_start),
-            ("date", "<=", period_date_stop),
-            ("intrastat", "=", True),
-        ]
         inv_type = []
         if self.sale:
             inv_type += ["out_invoice", "out_refund"]
         if self.purchase:
             inv_type += ["in_invoice", "in_refund"]
-        domain.append(("move_type", "in", inv_type))
+
+        domain = [
+            "&",
+            ("invoice_id.intrastat", "=", True),
+            ("invoice_id.move_type", "in", inv_type),
+            "|",
+            "&",
+            ("operation_date", "=", False),
+            "&",
+            ("invoice_id.date", ">=", period_date_start),
+            ("invoice_id.date", "<=", period_date_stop),
+            "&",
+            ("operation_date", "!=", False),
+            "&",
+            ("operation_date", ">=", period_date_start),
+            ("operation_date", "<=", period_date_stop),
+        ]
```

Comment:
Il dominio che stai facendo qui è:
(A and B) and ((C and D and E) or (F and G and H))
che sarebbe:
            &
      /            \
    &              |
  /    \         /      \
A      B    &         &
             /   \       /  \
           C     &   F    &
                 /  \         / \
               D    E     G   H
Che diventerebbe:
& & A B | & C & D E & F & G H
mentre qui è:
& A B | & C & D E & F & G H
Mi sono perso qualcosa? (non sono mai stato il massimo con questa roba ;))

Reply:
Quindi manca un `&` all'inizio?
In pratica dovrebbe essere equivalente perchè in mancanza di operatore viene usato implicitamente `&`.
Lo rendiamo più esplicito?

Reply:
Sarei per essere esplicito, se non è una prassi.